README.md

```diff
@@ -1,45 +1,601 @@
+}
+
+public class PaymentMethod {
+  private PaymentMethodType paymentMethodType = null;
+}
+
+public enum PaymentMethodType {
+  CARD("CARD"),
+  ALIPAY("ALIPAY"),
+  WECHAT("WECHAT")
+}
 ```
 
-## Samples
+Request parameters:
+- `store_id` - the ID of your assigned store
+- `payment_reference` - a unique reference for the payment, sent from the client and returned by the server
+- `amount` - the amount to request for the online payment, formatted as a 2.dp decimal number, such as `1.00`
+- `currency` - the currency for this payment in 3 character ISO format, such as `GBP`
+- `order_id` - a unique order ID for this payment, sent from the client and returned by the server
+- `return_url` - where the payment form will redirect to after the payment has completed, the return url will have additional query parameters appended to indicate the status of the payment request.
+- `showTimer` - flag to show countdown timer in payment page
+- `expiry` - how long the payment form will wait until the payment expires and the page will redirect to the return url
+
+
+#### PayResponse : Payment Response
+
+````java
+public enum ErrorType {
+  UNKNOWN("UNKNOWN"),
+  DUPLICATE_ATTEMPT("DUPLICATE_ATTEMPT"),
+  INVALID_REQUEST("INVALID_REQUEST"),
+  CANCELLED("CANCELLED")
+}
+
+public class Error {
+  private ErrorType type;       //Enum for the error type
+  private String message;       //Error message
+}
+
+public class PaymentInitiationResponse {
+    private Response response;
+    private Error error;
+}
+
+public class Response {
+  private String paymentId;     // Unique identifier created by Kody
+  private String paymentUrl;    // The URL to send to the user from your application 
+}
+````
+
+#### Java Demo
+````java
+import common.PaymentClient;
+import com.kodypay.grpc.pay.v1.*;
+
+PaymentClient paymentClient = new PaymentClient(HOSTNAME, APIKEY);
+
+String storeId = "UUID of assigned store";
+String orderId = "UUID of assigned store";
+String paymentReference = "payment reference";
+String amount = "1.00";
+String currency = "GBP";
+String returnUrl = "returnUrl";
+int expiringSeconds = 1800;
+
+PaymentInitiationRequest paymentInitiationRequest = PaymentInitiationRequest.newBuilder()
+        .setStoreId(storeId)
+        .setPaymentReference(paymentReference)
+        .setAmount(amount)
+        .setCurrency(currency)
+        .setOrderId(orderId)
+        .setReturnUrl(returnUrl)
+        .setExpiry(PaymentInitiationRequest.ExpirySettings.newBuilder()
+                .setShowTimer(true)
+                .setExpiringSeconds(expiringSeconds)
+                .build()
+        ).build();
+
+PaymentInitiationResponse response = paymentClient.sendOnlinePayment(paymentInitiationRequest);
+````
+
+### Get Online Payments
+
+The payment details request requires the following parameters:
+- `storeId` - the ID of your assigned store
+- `pageCursor` - the Order ID returned in the initial payment response, a unique UUID value for each payment.
```

Comment:
Is that the correct description?
What is a PageCursor ? No example of it is given below

Reply:
```
PageCursor:
  PageSize (1)
```